src/zenml/materializers/path_materializer.py

```diff
@@ -0,0 +1,643 @@
+        """Check if a file is likely a text file based on extension or content.
+
+        Args:
+            file_path: Path to the file to check
+
+        Returns:
+            True if the file is likely a text file, False otherwise
+        """
+        # Common text file extensions
+        text_extensions = {
+            ".txt",
+            ".log",
+            ".md",
+            ".markdown",
+            ".csv",
+            ".json",
+            ".yaml",
+            ".yml",
+            ".html",
+            ".htm",
+            ".css",
+            ".js",
+            ".py",
+            ".java",
+            ".c",
+            ".cpp",
+            ".h",
+            ".sh",
+            ".bat",
+            ".cmd",
+            ".xml",
+            ".sql",
+            ".diff",
+            ".patch",
+            ".gitignore",
+        }
+
+        if file_path.suffix.lower() in text_extensions:
+            return True
+
+        # For files without recognized extensions, try to detect if they're text
+        try:
+            # Read the first 1024 bytes and check if they're valid UTF-8
+            with open(file_path, "rb") as f:
+                sample = f.read(1024)
+            sample.decode("utf-8")
+            return True
+        except (UnicodeDecodeError, IOError):
+            return False
+
+    def _format_size(self, size_bytes: int) -> str:
+        """Format file size in a human-readable way.
+
+        Args:
+            size_bytes: Size in bytes
+
+        Returns:
+            Formatted size string (e.g., "1.23 MB")
+        """
+        units = ["B", "KB", "MB", "GB", "TB"]
+        size = float(size_bytes)
+        unit_index = 0
+
+        while size >= 1024.0 and unit_index < len(units) - 1:
+            size /= 1024.0
+            unit_index += 1
+
+        return f"{size:.1f} {units[unit_index]}"
+
+    def _create_direct_download_html(
+        self, files_data: list[Dict[str, Any]]
+    ) -> str:
+        """Create an HTML page with direct download links.
+
+        Instead of using JavaScript to create downloads, this approach
+        uses direct data URIs in anchor tags, which should be more compatible
+        with the ZenML dashboard environment.
+
+        Args:
+            files_data: List of dictionaries with file information
+
+        Returns:
+            HTML content as a string
+        """
+        # Create file rows with direct download links
+        file_rows = ""
+        for file_info in files_data:
+            # Create a direct download link with appropriate MIME type
+            mime_type = self._get_mime_type(file_info["path"])
+            # Text files work better with text/plain MIME type for downloads
+            download_mime_type = (
+                "text/plain" if file_info["is_text"] else mime_type
+            )
+            download_button = f"""
+            <a href="data:{download_mime_type};base64,{file_info["data"]}" 
```

Comment:
How is this even supposed to work for any directory with some larger files?

Reply:
@schustmi yeah it doesnt work, i htink we need to limit it